File: sdk/python/coreauth/types/connections.py
"""Connection types."""

from __future__ import annotations

from typing import Any, Optional

from pydantic import BaseModel


class Connection(BaseModel):
    id: str
    name: str
    connection_type: str
    scope: str
    organization_id: Optional[str] = None
    config: Any = None
    is_enabled: bool = True
    created_at: Optional[str] = None
    updated_at: Optional[str] = None


class CreateConnectionRequest(BaseModel):
    name: str
    connection_type: str
    config: Any = None


class UpdateConnectionRequest(BaseModel):
    name: Optional[str] = None
    config: Any = None
    is_enabled: Optional[bool] = None


class AuthMethod(BaseModel):
    connection_id: str
    name: str
    method_type: str
    scope: str

File: sdk/node/src/types/connections.ts
export interface Connection {
  id: string;
  name: string;
  connection_type: string;
  scope: string;
  organization_id?: string;
  config?: Record<string, unknown>;
  is_enabled: boolean;
  created_at?: string;
  updated_at?: string;
}

export interface CreateConnectionRequest {
  name: string;
  connection_type: string;
  config?: Record<string, unknown>;
}

export interface UpdateConnectionRequest {
  name?: string;
  config?: Record<string, unknown>;
  is_enabled?: boolean;
}

export interface AuthMethod {
  connection_id: string;
  name: string;
  method_type: string;
  scope: string;
}
